### PR TITLE
[LUPEYALPHA-1004] New status for FE claims: Awaiting provider verification

### PR DESCRIPTION
**Pull request by alkesh**
https://dfedigital.atlassian.net/browse/LUPEYALPHA-1004

### CODE REVIEW

File: app/models/claim.rb [hunk 1]
```diff
@@ -431,8 +433,24 @@ def same_claimant?(other_claim)
     end
   end
 
+  def awaiting_provider_verification?
+    return false unless has_fe_policy?
+
+    return false if eligibility.verification.present?
```
Comment (rjlynch) [inline app/models/claim.rb:439]: Slight nitpick but could we use
```suggestion
    return false if eligibility.verified?
```
so we keep the definition of "being verified" in one place?

File: app/models/claim.rb [hunk 2]
```diff
@@ -431,8 +433,24 @@ def same_claimant?(other_claim)
...
+
+    if tasks.where(name: "matching_details").any?
+      tasks.where(name: "matching_details", passed: true).any?
+    else
+      Claim::MatchingAttributeFinder.new(self).matching_claims.empty?
+    end
```
Comment (rjlynch) [inline app/models/claim.rb:445]: I know this is the logic described on the ticket but I wonder if what they _really_ want to know is if the provider verification email has been sent out and we're waiting on a response 🤔 if so we could move the logic from
`Policies::FurtherEducationPayments::AdminProviderVerificationTaskPresenter#verification_email_sent?` on to the eligibility then use that method here.

File: spec/support/admin_view_claim_feature_shared_examples.rb [hunk 3]
```diff
@@ -42,7 +42,11 @@
   }
 
   let!(:approved_awaiting_payroll_claim) {
-    eligibility = create(:"#{policy.to_s.underscore}_eligibility", :eligible)
+    eligibility = if policy == Policies::FurtherEducationPayments
+      create(:"#{policy.to_s.underscore}_eligibility", :verified)
+    else
+      create(:"#{policy.to_s.underscore}_eligibility", :eligible)
```
Comment (rjlynch) [inline spec/support/admin_view_claim_feature_shared_examples.rb:48]: Really hate shared example specs 🙃 
I was going to suggest aliasing `eligible` to `verified` in the FE eligibility factory but they kind of have different meanings

**Review by rjlynch — APPROVED** [review]
Looks good to me, left one comment that might be worth checking with the product team to see what they're trying to achieve with the new status

File: app/models/claim.rb (hunk 2 above)
Comment (alkesh) [reply app/models/claim.rb:457]: I've asked on the ticket. If we change to using `verification_email_sent`, then the scope `awaiting_further_education_provider_verification` will have to change to check notes (I have a query ready that works).

**Review by rjlynch — APPROVED** [review]
Looks good 👍 
I might be tempted to put a scope on FE eligibility `awaiting_provider_verification` and then simplify the scope on the `Claim` to just merge the FE scope in, but I think it's fine as it

File: spec/support/admin_view_claim_feature_shared_examples.rb (hunk 3 above)
Comment (alkesh) [reply spec/support/admin_view_claim_feature_shared_examples.rb:48]: I think you're right.
The `eligible` trait should also be verified - I'm going to make a separate trait of `not_verified` for the other tests.

**Pull request merged**